<?xml version="1.0"?>
<launch>
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find agv)/config/rosconsole.config"/>

  <node name="collision_detector" pkg="agv" type="collision_detector" output="screen">
    <param name="odom_topic"                  value="/odometry/filtered"/>
    <param name="obstacle_topic"              value="/obstacles"/>
    <param name="obstacle2_topic"             value="/obstacles2"/>
    <param name="obstacle3_topic"             value="/obstacles3"/>
    <param name="steering_angle_topic"        value="/present_steering_angle"/>

    <param name="collision_detector"          value="/collision_detector"/>
    <param name="predicted_trajectory_topic"  value="/predicted_trajectory"/>
    <param name="dynamic_obstacle_path_topic" value="/dynamic_obstacle_path_topic"/>
    <param name="time_to_collision_topic"     value="/time_to_collision"/>

    <param name="planning_frequency"          value="10"/>
    <param name="collision_time_threshold"    value="5"/>
    <param name="look_ahead_time"             value="5"/>
    <param name="delta_t"                     value="0.5"/>
    <param name="circle_radius"               value="0.7"/>
    <param name="safety_margin"               value="0.3"/>
    <param name="back_safety_margin"          value="0.15"/>
    <param name="speed_threshold"             value="0.3"/>
    <param name="dynamic_obstacle_speed"      value="0.3"/>

    <param name="baselink_to_front_axle"      value="2.2"/>
    <param name="cog_to_front_axle"           value="1.1"/>
   </node>

  <node name="lane_publisher" pkg="agv" type="lane_publisher" output="screen">
    <param name="odom_topic"                  value="/odometry/filtered"/>
    <param name="path_filename_topic"         value="/path_filename_topic"/>

    <param name="lane_info_topic"             value="/lane_info_topic"/>
    <param name="left_boundary_xa_topic"      value="/left_boundary_xa_topic"/>
    <param name="left_boundary_ax_topic"      value="/left_boundary_ax_topic"/>
    <param name="right_boundary_topic"        value="/right_boundary_topic"/>
    <param name="centerline_topic"            value="/centerline_topic"/>
    <param name="waypoints_info_topic"        value="/waypoints_info_topic"/>
    <param name="special_waypoint"            value="/special_waypoint"/>
    
    <param name="planning_frequency"          value="10"/>
    <param name="lanes_directory"             value="$(find agv)/lanes/"/>
    <param name="near_turning_point_radius"   value="9.0"/>
    <param name="end_point_stop_radius"       value="7.0"/>
    <param name="end_point_slowdown_radius"   value="15"/>
  </node>

</launch>